draft-ietf-taps-interface.md

```diff
@@ -1594,114 +1309,887 @@ does provide the following guarantees about the ordering of operations:
+: The classification in this version of the draft emphasizes the way the property effects the transport system and protocols. It treats the aspect of whether properties are used on a connection, preconnection or message as an orthogonal dimension of classification.
+: The "Message Properties" from I-D.ietf-taps-interface-00 therefore have been split into "Protocol Properties" – emphasizing that they effect the protocol configurations – and "Control Properties" – emphasizing that they control the local transport system itself.
+
+
+### Selection Properties {#selection-props}
+
+Selection Properties influence protocol and path selection.
+Their value usually is or includes a Preference that constrains (in case of
+Require or Prohibit) or influences (Prefer, Ignore, Avoid) the selection of
+transport protocols and paths used.
+
+An implementation of this interface must provide sensible defaults for
+Selection Properties. The defaults given for each property below represent a
+configuration that can be implemented over TCP. An alternate set of default
+Protocol Selection Properties would represent a configuration that can be
+implemented over UDP.
+
+Protocol Selection Properties can only be set on Preconnections, see {{connection-props}}.
+Path Selection Properties are usually used on Preconnections, but
+might also be used on massages to assist per-message path selection for
+multipath aware protocols.
+
+
+### Protocol Properties {#protocol-props}
+
+Protocol Properties represent the configuration of the selected Protocol Stacks
+backing a Connection. Some properties apply generically across multiple
+transport protocols, while other properties only apply to specific protocols.
+Generic properties will be passed to the selected candidate Protocol Stack(s) 
+to configure them before candidate Connection establishment.
+The default settings of these properties will vary based on the specific
+protocols being used and the system's configuration.
+
+Most Protocol Properties can be set on a Preconnection during pre-establishment to preconfigure Protocol Stacks during establishment.
+
+In order to specify Specific Protocol Properties, Transport System
+implementations may offer applications to attach a set of options to the
+Preconnection Object, associated with a specific protocol. For example, an
+application could specify a set of TCP Options to use if and only if TCP is
+selected by the system. Such properties must not be assumed to apply across
+different protocols. Attempts to set specific protocol properties on a
+protocol stack not containing that specific protocol are simply ignored, and
+do not raise an error.
+
+Note that many protocol properties have a corresponding selection property
+which asks for a protocol providing a specific transport feature that is
+controlled by the protocol property.
+
+### Control Properties {#control-props}
+
+\[TODO: Discuss]
+
+Control properties control the local transport system behavior or request 
```

Comment:
"Control properties control" is a bit redundant

Reply:
True. Changed it to "manage", not perfect but hopefully still works for what is meant here.